Gems/PhysX/Code/Source/Articulation.cpp

```diff
@@ -62,50 +62,44 @@ namespace PhysX
 
     void ArticulationLink::AddCollisionShape(const ArticulationLinkData& thisLinkData)
     {
-        const Physics::ColliderConfiguration* colliderConfiguration = thisLinkData.m_shapeColliderConfiguration.first.get();
-        const Physics::ShapeConfiguration* shapeConfiguration = thisLinkData.m_shapeColliderConfiguration.second.get();
-
-        if (shapeConfiguration && colliderConfiguration)
+        m_physicsShapes.clear();
+        for (auto &[colliderConfiguration, shapeConfiguration]: thisLinkData.m_shapeColliderConfigurationList)
         {
-            if (shapeConfiguration->GetShapeType() == Physics::ShapeType::PhysicsAsset)
+            if (shapeConfiguration && colliderConfiguration)
             {
-                const auto* physicsAssetShapeConfiguration =
-                    static_cast<const Physics::PhysicsAssetShapeConfiguration*>(shapeConfiguration);
-                if (!physicsAssetShapeConfiguration->m_asset.IsReady())
+                if (shapeConfiguration->GetShapeType() == Physics::ShapeType::PhysicsAsset)
                 {
-                    const_cast<Physics::PhysicsAssetShapeConfiguration*>(physicsAssetShapeConfiguration)->m_asset.BlockUntilLoadComplete();
+                    const auto* physicsAssetShapeConfiguration =
+                        static_cast<const Physics::PhysicsAssetShapeConfiguration*>(shapeConfiguration.get());
+                    if (!physicsAssetShapeConfiguration->m_asset.IsReady())
+                    {
+                        const_cast<Physics::PhysicsAssetShapeConfiguration*>(physicsAssetShapeConfiguration)->m_asset.BlockUntilLoadComplete();
```

Comment:
Shouldn't this check the reutrn value of BlockUntilLoadComplete, to verify that it returns `AZ::AssetData::AssetStatus::Ready`

Reply:
even though this is how it was done previously, can this be updated in the future to not require blocking until load complete and use asset ready and reloaded notifications instead?

Reply:
@lumberyard-employee-dm thanks for suggestion added check and error message.

@gadams3 I would prefer to treat articulation as a critical asset. By design in PhysX 5 and 4 you cannot change the hierarchy and structure of articulation during runtime, so reloading prefabs with Articulations will not work as expected.